Description: Disable tests that fail locally
diff --git a/test/parallel/parallel.status b/test/parallel/parallel.status
index 2f3d912a9..e2f9802d3 100644
--- a/test/parallel/parallel.status
+++ b/test/parallel/parallel.status
@@ -17,6 +17,12 @@ test-fs-rmdir-recursive: PASS, FLAKY
 [$system==linux]
 # https://github.com/nodejs/node/issues/39368
 test-domain-error-types: PASS,FLAKY
+# fail locally
+test-cluster-bind-privileged-port: PASS,FLAKY
+test-cluster-shared-handle-bind-privileged-port: PASS,FLAKY
+test-module-loading-globalpaths: PASS,FLAKY
+test-repl: PASS,FLAKY
+test-cluster-dgram-ipv6only: PASS,FLAKY
 
 [$system==macos]
 # https://github.com/nodejs/node/issues/42741
