Description: allow vfp2 and allow setting arm_version option
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2022-08-29
Forwarded: https://github.com/nodejs/node/issues/44357
--- a/configure.py
+++ b/configure.py
@@ -51,7 +51,7 @@
 valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64el', 'ppc',
               'ppc64', 'x64', 'x86', 'x86_64', 's390x', 'riscv64', 'loong64')
 valid_arm_float_abi = ('soft', 'softfp', 'hard')
-valid_arm_fpu = ('vfp', 'vfpv3', 'vfpv3-d16', 'neon')
+valid_arm_fpu = ('vfp', 'vfpv2', 'vfpv3', 'vfpv3-d16', 'neon')
 valid_mips_arch = ('loongson', 'r1', 'r2', 'r6', 'rx')
 valid_mips_fpu = ('fp32', 'fp64', 'fpxx')
 valid_mips_float_abi = ('soft', 'hard')
@@ -540,6 +540,13 @@
     choices=valid_arm_fpu,
     help=f"ARM FPU mode ({', '.join(valid_arm_fpu)}) [default: %(default)s]")
 
+parser.add_argument('--with-arm-version',
+    action='store',
+    dest='arm_version',
+    default=None,
+    help='ARM Version'
+)
+
 parser.add_argument('--with-mips-arch-variant',
     action='store',
     dest='mips_arch_variant',
@@ -1215,8 +1222,9 @@
     arm_float_abi = 'default'
 
   arm_fpu = 'vfp'
-
-  if is_arch_armv7():
+  if options.arm_version:
+    o['variables']['arm_version'] = options.arm_version
+  elif is_arch_armv7():
     arm_fpu = 'vfpv3'
     o['variables']['arm_version'] = '7'
   else:
