Index: nodejs/test/common/index.js
===================================================================
--- nodejs.orig/test/common/index.js
+++ nodejs/test/common/index.js
@@ -76,7 +76,8 @@ const hasOpenSSL = (major = 0, minor = 0
   return OPENSSL_VERSION_NUMBER >= opensslVersionNumber(major, minor, patch);
 };
 
-const hasQuic = hasCrypto && !!process.config.variables.openssl_quic;
+// Pexip: we don't have Quic
+const hasQuic = false;
 
 function parseTestFlags(filename = process.argv[1]) {
   // The copyright notice is relatively big and the flags could come afterwards.
