Index: nodejs/src/crypto/crypto_dsa.cc
===================================================================
--- nodejs.orig/src/crypto/crypto_dsa.cc
+++ nodejs/src/crypto/crypto_dsa.cc
@@ -88,7 +88,7 @@ Maybe<bool> DsaKeyGenTraits::AdditionalC
 
   params->params.modulus_bits = args[*offset].As<Uint32>()->Value();
   params->params.divisor_bits = args[*offset + 1].As<Int32>()->Value();
-  CHECK_GE(params->params.divisor_bits, -1);
+  CHECK_GE(params->params.divisor_bits, 0);
 
   *offset += 2;
 
@@ -138,10 +138,15 @@ Maybe<bool> GetDsaKeyDetail(
   int type = EVP_PKEY_id(m_pkey.get());
   CHECK(type == EVP_PKEY_DSA);
 
+#if OPENSSL_VERSION_MAJOR >= 3
+  CHECK_GE(EVP_PKEY_get_bn_param(key, OSSL_PKEY_PARAM_FFC_P, &p), 1);
+  CHECK_GE(EVP_PKEY_get_bn_param(key, OSSL_PKEY_PARAM_FFC_Q, &q), 1);
+#else
   const DSA* dsa = EVP_PKEY_get0_DSA(m_pkey.get());
   CHECK_NOT_NULL(dsa);
 
   DSA_get0_pqg(dsa, &p, &q, nullptr);
+#endif
 
   size_t modulus_length = BN_num_bits(p);
   size_t divisor_length = BN_num_bits(q);
Index: nodejs/src/crypto/crypto_sig.cc
===================================================================
--- nodejs.orig/src/crypto/crypto_sig.cc
+++ nodejs/src/crypto/crypto_sig.cc
@@ -34,15 +34,17 @@ bool ValidateDSAParameters(EVP_PKEY* key
 #if OPENSSL_VERSION_MAJOR >= 3
   if (EVP_default_properties_is_fips_enabled(nullptr) &&
       EVP_PKEY_DSA == EVP_PKEY_base_id(key)) {
+    const BIGNUM* p, q;
+    CHECK_GE(EVP_PKEY_get_bn_param(key, OSSL_PKEY_PARAM_FFC_P, &p), 1);
+    CHECK_GE(EVP_PKEY_get_bn_param(key, OSSL_PKEY_PARAM_FFC_Q, &q), 1);
 #else
   if (FIPS_mode() && EVP_PKEY_DSA == EVP_PKEY_base_id(key)) {
-#endif
     const DSA* dsa = EVP_PKEY_get0_DSA(key);
-    const BIGNUM* p;
+    const BIGNUM* p, q;
     DSA_get0_pqg(dsa, &p, nullptr, nullptr);
-    size_t L = BN_num_bits(p);
-    const BIGNUM* q;
     DSA_get0_pqg(dsa, nullptr, &q, nullptr);
+#endif
+    size_t L = BN_num_bits(p);
     size_t N = BN_num_bits(q);
 
     return (L == 1024 && N == 160) ||
@@ -126,9 +128,15 @@ unsigned int GetBytesOfRS(const ManagedE
   int bits, base_id = EVP_PKEY_base_id(pkey.get());
 
   if (base_id == EVP_PKEY_DSA) {
+#if OPENSSL_VERSION_MAJOR >= 3
+    const BIGNUM* q;
+    CHECK_GE(EVP_PKEY_get_bn_param(key, OSSL_PKEY_PARAM_FFC_Q, &q), 1);
+    bits = BN_num_bits(q);
+#else
     const DSA* dsa_key = EVP_PKEY_get0_DSA(pkey.get());
     // Both r and s are computed mod q, so their width is limited by that of q.
     bits = BN_num_bits(DSA_get0_q(dsa_key));
+#endif
   } else if (base_id == EVP_PKEY_EC) {
     const EC_KEY* ec_key = EVP_PKEY_get0_EC_KEY(pkey.get());
     const EC_GROUP* ec_group = EC_KEY_get0_group(ec_key);
