Description: build using ada upstream tarball component
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2023-11-30
Forwarded: not-needed

--- a/deps/ada/ada.gyp
+++ b/deps/ada/ada.gyp
@@ -1,15 +1,15 @@
 {
   'variables': {
     'v8_enable_i18n_support%': 1,
-    'ada_sources': [ 'ada.cpp' ],
+    'ada_sources': [ '../../ada/src/ada.cpp' ],
   },
   'targets': [
     {
       'target_name': 'ada',
       'type': 'static_library',
-      'include_dirs': ['.'],
+      'include_dirs': ['../../ada/include'],
       'direct_dependent_settings': {
-        'include_dirs': ['.'],
+        'include_dirs': ['../../ada/include'],
       },
       'sources': [ '<@(ada_sources)' ]
     },
