From f347c53855f82d60acbce2b24b4c01436ee09af2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C3=ABl=20Zasso?= <targos@protonmail.com>
Date: Thu, 20 Oct 2022 13:39:07 +0200
Subject: [PATCH] fixup! fixup! fixup! deps: update ICU to 72.1

---
 test/parallel/test-intl.js | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/test/parallel/test-intl.js
+++ b/test/parallel/test-intl.js
@@ -97,7 +97,12 @@
   // Test format
   {
     const localeString = date0.toLocaleString(['en'], optsGMT);
-    assert.strictEqual(localeString, '1/1/1970, 12:00:00 AM');
+    if (Number(process.versions.cldr) >= 42) {
+      assert.strictEqual(localeString, '1/1/1970, 12:00:00 AM');
+    } else {
+      assert.strictEqual(localeString, '1/1/1970, 12:00:00 AM');
+    }
+
   }
   // number format
   {
