Description: add acorn, walk to shared builtins
Last-Update: 2022-09-28
Forwarded: https://github.com/nodejs/node/pull/44376
Author: Jérémy Lal <kapouer@melix.org>

--- a/configure.py
+++ b/configure.py
@@ -59,7 +59,9 @@
 
 shareable_builtins = {'cjs_module_lexer/lexer': 'deps/cjs-module-lexer/lexer.js',
                      'cjs_module_lexer/dist/lexer': 'deps/cjs-module-lexer/dist/lexer.js',
-                     'undici/undici': 'deps/undici/undici.js'
+                     'undici/undici': 'deps/undici/undici.js',
+                     'acorn': 'deps/acorn/acorn/dist/acorn.js',
+                     'acorn_walk': 'deps/acorn/acorn-walk/dist/walk.js'
 }
 
 # create option groups
--- a/src/node_builtins.cc
+++ b/src/node_builtins.cc
@@ -49,6 +49,19 @@
   AddExternalizedBuiltin("internal/deps/undici/undici",
                          STRINGIFY(NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH));
 #endif  // NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH
+
+#ifdef NODE_SHARED_BUILTIN_ACORN_PATH
+  AddExternalizedBuiltin(
+    "internal/deps/acorn/acorn/dist/acorn",
+    STRINGIFY(NODE_SHARED_BUILTIN_ACORN_PATH));
+#endif  // NODE_SHARED_BUILTIN_ACORN_PATH
+
+#ifdef NODE_SHARED_BUILTIN_ACORN_WALK_PATH
+  AddExternalizedBuiltin(
+    "internal/deps/acorn/acorn-walk/dist/walk",
+    STRINGIFY(NODE_SHARED_BUILTIN_ACORN_WALK_PATH));
+#endif  // NODE_SHARED_BUILTIN_ACORN_WALK_PATH
+
 #endif  // NODE_HAVE_I18N_SUPPORT
 }
 
--- a/node.gyp
+++ b/node.gyp
@@ -49,8 +49,6 @@
       'deps/v8/tools/tickprocessor.mjs',
       'deps/v8/tools/sourcemap.mjs',
       'deps/v8/tools/tickprocessor-driver.mjs',
-      'deps/acorn/acorn/dist/acorn.js',
-      'deps/acorn/acorn-walk/dist/walk.js',
       '<@(node_builtin_shareable_builtins)',
     ],
     'node_mksnapshot_exec': '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)node_mksnapshot<(EXECUTABLE_SUFFIX)',
