Description: skip useless files in build target
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: https://github.com/nodejs/node/issues/55787

--- a/Makefile
+++ b/Makefile
@@ -171,7 +171,7 @@
 	$(warning '$@' target is a noop)
 
 out/Makefile: config.gypi common.gypi node.gyp \
-	deps/uv/uv.gyp deps/llhttp/llhttp.gyp deps/zlib/zlib.gyp \
+	deps/llhttp/llhttp.gyp \
 	deps/simdutf/simdutf.gyp deps/ada/ada.gyp \
 	tools/v8_gypfiles/toolchain.gypi tools/v8_gypfiles/features.gypi \
 	tools/v8_gypfiles/inspector.gypi tools/v8_gypfiles/v8.gyp
