Description: fix double register usage on mipsel
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-yet
Last-Update: 2022-06-15
--- a/deps/v8/src/codegen/mips/register-mips.h
+++ b/deps/v8/src/codegen/mips/register-mips.h
@@ -32,9 +32,8 @@
 // Currently, MIPS just use even float point register, except
 // for C function param registers.
 #define DOUBLE_USE_REGISTERS(V)                           \
-  V(f0)  V(f2)  V(f4)  V(f6)  V(f8)  V(f10) V(f12) V(f13) \
-  V(f14) V(f15) V(f16) V(f18) V(f20) V(f22) V(f24) V(f26) \
-  V(f28) V(f30)
+  V(f0)  V(f2)  V(f4)  V(f6)  V(f8)  V(f10) V(f12) V(f14) \
+  V(f16) V(f18) V(f20) V(f22) V(f24) V(f26) V(f28) V(f30)
 
 #define FLOAT_REGISTERS DOUBLE_REGISTERS
 #define SIMD128_REGISTERS(V)                              \
