Description: mipsel in debian supports 32-bit processors
Forwarded: not-needed
Last-Update: 2021-11-03
Author: Jérémy Lal <kapouer@melix.org>
--- a/test/common/index.js
+++ b/test/common/index.js
@@ -33,7 +33,7 @@
 const { isMainThread } = require('worker_threads');
 
 const tmpdir = require('./tmpdir');
-const bits = ['arm64', 'mips', 'mipsel', 'ppc64', 'riscv64', 's390x', 'x64']
+const bits = ['arm64', 'mips64el', 'ppc64', 'riscv64', 's390x', 'x64']
   .includes(process.arch) ? 64 : 32;
 const hasIntl = !!process.config.variables.v8_enable_i18n_support;
 
