Subject: test does not pass on riscv64
Last-Update: 2022-12-14
Author: Jérémy Lal <kapouer@melix.org>, Bo YU <tsu.yubo@gmail.com>
Forwarded: not-yet

--- a/test/sequential/sequential.status
+++ b/test/sequential/sequential.status
@@ -61,3 +61,16 @@
 [$arch==s390x]
 # https://github.com/nodejs/node/issues/41286
 test-performance-eventloopdelay: PASS, FLAKY
+
+[$arch==riscv64]
+test-diagnostic-dir-cpu-prof: FLAKY
+test-cpu-prof-worker-argv: FLAKY
+test-cpu-prof-exit: FLAKY
+test-cpu-prof-kill: FLAKY
+test-diagnostic-dir-cpu-prof: FLAKY
+test-debugger-auto-resume: FLAKY
+test-debugger-preserve-breaks: FLAKY
+test-cpu-prof-dir-relative: FLAKY
+test-cpu-prof-drained: FLAKY
+test-worker-prof: FLAKY
+test-vm-timeout-escape-promise-module-2: FLAKY
--- a/test/parallel/parallel.status
+++ b/test/parallel/parallel.status
@@ -127,3 +127,11 @@
 test-http-upgrade-advertise: PASS, FLAKY
 test-tls-client-mindhsize: PASS, FLAKY
 test-tls-write-error: PASS, FLAKY
+
+[$arch==riscv64]
+test-fetch: PASS, FLAKY
+test-net-socket-connect-without-cb: PASS, FLAKY
+test-runner-inspect: PASS, FLAKY
+test-tcp-wrap-listen: PASS, FLAKY
+test-wasm-web-api: PASS, FLAKY
+test-vm-timeout-escape-promise-module: PASS, FLAKY
