Subject: test does not pass on riscv64
Last-Update: 2022-12-14
Author: Jérémy Lal <kapouer@melix.org>, Bo YU <tsu.yubo@gmail.com>
Forwarded: not-yet

--- a/test/sequential/sequential.status
+++ b/test/sequential/sequential.status
@@ -65,3 +65,19 @@
 [$arch==s390x]
 # https://github.com/nodejs/node/issues/41286
 test-performance-eventloopdelay: PASS, FLAKY
+
+[$arch==riscv64]
+test-diagnostic-dir-cpu-prof: PASS, FLAKY
+test-cpu-prof-worker-argv: PASS, FLAKY
+test-cpu-prof-exit: PASS, FLAKY
+test-cpu-prof-kill: PASS, FLAKY
+test-diagnostic-dir-cpu-prof: PASS, FLAKY
+test-debugger-auto-resume: PASS, FLAKY
+test-debugger-preserve-breaks: PASS, FLAKY
+test-cpu-prof-dir-relative: PASS, FLAKY
+test-cpu-prof-drained: PASS, FLAKY
+test-worker-prof: PASS, FLAKY
+test-vm-timeout-escape-promise-module-2: PASS, FLAKY
+test-cpu-prof-dir-absolute: PASS, FLAKY
+
+
--- a/test/parallel/parallel.status
+++ b/test/parallel/parallel.status
@@ -152,6 +152,22 @@
 # https://github.com/nodejs/node/issues/48047
 test-http-pipeline-flood: SKIP
 
+[$arch==riscv64]
+test-fetch: PASS, FLAKY
+test-net-socket-connect-without-cb: PASS, FLAKY
+test-runner-inspect: PASS, FLAKY
+test-tcp-wrap-listen: PASS, FLAKY
+test-wasm-web-api: PASS, FLAKY
+test-vm-timeout-escape-promise-module: PASS, FLAKY
+test-crypto-sign-verify: PASS, FLAKY
+test-freeze-intrinsics: PASS, FLAKY
+test-tls-root-certificates: PASS, FLAKY
+test-vm-global-non-writable-properties: PASS, FLAKY
+test-vm-global-setter: PASS, FLAKY
+test-vm-module-synthetic: PASS, FLAKY
+test-vm-strict-assign: PASS, FLAKY
+
+
 [$asan==on]
 # https://github.com/nodejs/node/issues/39655
 test-cluster-primary-error: PASS, FLAKY
